src/com/amazon/corretto/crypto/provider/EvpSignature.java

```diff
@@ -348,20 +363,29 @@ private EvpSignature(AmazonCorrettoCryptoProvider provider, final EvpKeyType key
             .withSinglePass((src, offset, length) ->
                 key_.use(ptr ->
                     sign(ptr,
-                            digestName_, paddingType, null, 0,
+                            digestName_, paddingType_, pssMgfMd, pssSaltLen,
                             src, offset, length))
             );
-        verifyingBuffer = new InputBuffer<Boolean, EvpContext, SignatureException>(1024)
+    }
+
+    private InputBuffer<Boolean, EvpContext, SignatureException> getVerifyingBuffer() {
+        final String pssMgfMd = pssParams_ != null
```

Comment:
Ditto here on thinking the ternary operator makes things hard to read.

Reply:
sure.